src/CompositeReferenceFieldManager.php

```diff
@@ -78,10 +89,58 @@ public function entityDelete(EntityInterface $entity, FieldDefinitionInterface $
     if (!$this->isCompositeField($field_definition)) {
       return;
     }
-    $referenced_entities = $entity->get($field_definition->getName())->referencedEntities();
+
+    // When composite revisions config is set, we need to query all past
+    // revisions for entities that were referenced.
+    if ($this->isCompositeRevisionsField($field_definition)) {
+      // Get the revision table name dedicated for the given field definition
+      // for the query.
+      $field_storage_definition = $field_definition->getFieldStorageDefinition();
+      $entity_storage = $this->entityTypeManager->getStorage($entity->getEntityTypeId());
+      $table_mapping = $entity_storage->getTableMapping();
+      $table_name = $table_mapping->getDedicatedRevisionTableName($field_storage_definition);
+
+      // Check if the field has a dedicated revision table otherwise we query
+      // the revision data table of the entity.
+      if ($this->database->schema()->tableExists($table_name)) {
+        $target_id_field = $field_definition->getName() . '_' . $field_storage_definition->getMainPropertyName();
+        $query = $this->database->select($table_name, 'r')
+          ->fields('r', [$target_id_field])
+          ->condition('entity_id', $entity->id())
+          ->groupBy($target_id_field);
+      }
+      else {
+        // Get the entity revision data table name from the entity type
+        // definition for the query.
+        $entity_type_definition = $this->entityTypeManager->getDefinition($entity->getEntityTypeId());
+        $table_name = $this->entityTypeManager->getDefinition($entity->getEntityTypeId())->getRevisionDataTable();
+        // Construct the column name from the field and the main property name.
+        $target_id_field = $field_definition->getName() . '__' . $field_storage_definition->getMainPropertyName();
```

Comment:
```suggestion
        $target_id_field = $field_definition->getName() . '_' . $field_storage_definition->getMainPropertyName();
```
But actually it is duplication of the code above.

Reply:
Have you tried your proposal? It would fail, since the entity field data table uses double _ a separation.

Reply:
We clarified it in a call.